indico/modules/events/timetable/legacy.py

```diff
@@ -36,6 +39,11 @@ def __init__(self, management=False):
         self.management = management
 
     def serialize_timetable(self, event, days=None, hide_weekends=False, strip_empty_days=False):
+        # pre-fetch ContributionPrincipals (to save time)
+        principal_map = {contrib.id: contrib.acl_entries for contrib in Contribution.find(
```

Comment:
probably more readable if you assign the query first and then iterate over it in the list comprehension